addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone_test.go

```diff
@@ -394,17 +441,142 @@ var _ = Describe("ClusterbootstrapClone", func() {
+				Expect(updatedClusterBootstrap.Spec.AdditionalPackages[idx].ValuesFrom).To(Equal(fakeClusterBootstrapTemplate.Spec.AdditionalPackages[idx].ValuesFrom))
+			}
+		})
+
+		It("should not add the fields which are meant to be skipped", func() {
+			antreaAPIGroup := antreaconfigv1alpha1.GroupVersion.Group
+			fakeCPIClusterBootstrapPackage := constructFakeClusterBootstrapPackageWithSecretRef()
+			fakeCSIClusterBootstrapPackage := constructFakeClusterBootstrapPackageWithInlineRef()
+			// Update fakeClusterBootstrapTemplate by adding a fake CPI and CSI package
+			fakeClusterBootstrapTemplate.Spec.CPI = fakeCPIClusterBootstrapPackage
+			fakeClusterBootstrapTemplate.Spec.CSI = fakeCSIClusterBootstrapPackage
+
+			clusterBootstrap := &v1alpha3.ClusterBootstrap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fake-clusterbootstrap",
+					Namespace: "fake-cluster-ns",
+					UID:       "uid",
+				},
+				Spec: &v1alpha3.ClusterBootstrapTemplateSpec{
+					// We do not expect this part to be overwritten to be what fakeClusterBootstrapTemplate has
+					CNI: &v1alpha3.ClusterBootstrapPackage{
+						RefName: "foo-antrea-clusterbootstrarp-package",
+						ValuesFrom: &v1alpha3.ValuesFrom{
+							ProviderRef: &corev1.TypedLocalObjectReference{
+								APIGroup: &antreaAPIGroup,
+								Kind:     "AntreaConfig",
+								Name:     "fooAntreaConfig",
+							},
+						},
+					},
+					CSI: &v1alpha3.ClusterBootstrapPackage{
+						RefName: "foo-vsphere-csi-clusterbootstrarp-package",
+						ValuesFrom: &v1alpha3.ValuesFrom{
+							Inline: map[string]interface{}{"should-not-be-updated": true},
+						},
+					},
+					AdditionalPackages: []*v1alpha3.ClusterBootstrapPackage{
+						{RefName: fakePinnipedCBPackageRefName},
+					},
+				},
+			}
+
+			err := helper.AddMissingSpecFieldsFromTemplate(fakeClusterBootstrapTemplate, clusterBootstrap, map[string]interface{}{"valuesFrom": nil})
+			Expect(err).NotTo(HaveOccurred())
+			updatedClusterBootstrap := clusterBootstrap
+			// CPI should be added to updatedClusterBootstrap
+			Expect(updatedClusterBootstrap.Spec.CPI).NotTo(BeNil())
+			// CPI's valuesFrom should be skipped
+			Expect(updatedClusterBootstrap.Spec.CPI.ValuesFrom).To(BeNil())
+			Expect(updatedClusterBootstrap.Spec.CPI.RefName).To(Equal(fakeClusterBootstrapTemplate.Spec.CPI.RefName))
+
+			Expect(len(updatedClusterBootstrap.Spec.AdditionalPackages)).To(Equal(len(fakeClusterBootstrapTemplate.Spec.AdditionalPackages)))
+			for idx, _ := range updatedClusterBootstrap.Spec.AdditionalPackages {
+				Expect(updatedClusterBootstrap.Spec.AdditionalPackages[idx].RefName).To(Equal(fakeClusterBootstrapTemplate.Spec.AdditionalPackages[idx].RefName))
+				Expect(updatedClusterBootstrap.Spec.AdditionalPackages[idx].ValuesFrom).To(BeNil())
+			}
+		})
+
+		It("should not add the fields which are meant to be skipped in additionalPackages", func() {
```

Comment:
Not sure how this test is different from above. This test is asserting additionalPackages which is covered above.